File: 0128_longestConsecutiveSequence/main.cpp
#include <vector>
#include <unordered_set>
#include <algorithm>

using namespace std;

class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> mySet(nums.begin(), nums.end());
        int res = 0;
        for(int i = 0; i < nums.size(); ++i)
        {
            if(mySet.find(nums[i]-1) != mySet.end())
            {
                int end = nums[i] + 1;
                while(mySet.find(end) != mySet.end())
                    ++end;
                res = max(res, end - nums[i]);
            }
        }
        return res;
    }
};

File: 0128_longestConsecutiveSequence/main.py
from typing import List

class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        res = 0
        num_set = set(nums)

        for val in num_set:
            if (val-1) not in num_set:
                end = val
                while (end+1) in num_set:
                    end += 1
                res = max(res, end-val+1)
        return res